source/common/upstream/eds.cc

```diff
@@ -121,6 +122,9 @@ void EdsClusterImpl::onConfigUpdate(const Protobuf::RepeatedPtrField<ProtobufWkt
     throw EnvoyException(fmt::format("Unexpected EDS cluster (expecting {}): {}", cluster_name_,
                                      cluster_load_assignment.cluster_name()));
   }
+  // Scrub original type information; we don't config dump endpoints today and
+  // this is significant memory overhead.
```

Comment:
You mean just adding all the recursive fields?

Reply:
Adding the original type information comes at a memory overhead cost, as discussed in the above thread regarding inlined endpoints.